services/api/src/util/auth.ts

```diff
@@ -105,7 +106,26 @@ export const getCredentialsForLegacyToken = async token => {
     throw new Error('Decoding token resulted in "null" or "undefined".');
   }
 
-  const { role = 'none', aud, sub, iss, iat } = decoded;
+  const { role = 'none', aud, sub, iss, iat, exp } = decoded;
+
+  // check the expiration on legacy tokens, reject them if necessary
+  const maxExpiry = getConfigFromEnv('LEGACY_EXPIRY_MAX', '3600') // 1hour default
+  const rejectLegacyExpiry = getConfigFromEnv('LEGACY_EXPIRY_REJECT', 'false') // don't reject intially, just log
+  if (exp) {
+    if ((parseInt(exp)-parseInt(iat)) > parseInt(maxExpiry)) {
+      const msg = `Legacy token (sub:${sub}; iss:${iss}) expiry ${(parseInt(exp)-parseInt(iat))} is greater than ${parseInt(maxExpiry)}`
+      logger.warn(msg);
+      if (rejectLegacyExpiry == "true") {
+        throw new Error(msg);
+      }
+    }
+  } else {
```

Comment:
This is checking that the token has a maximum expiration window (e.g., only valid for 60 seconds), but it doesn't check if the token has actually expired. This would still allow a token that has expired (`exp` is a date in the past) to be used as long as the max expiration time is within limits.

I think there would need to be three checks (along with the `rejectLegacyExpirary` flag): If the token is expired, deny. If the token is not expired and has a too long expiration window, deny. If the token does not have an expiration date, deny.

Reply:
Isn't that what happens here https://github.com/uselagoon/lagoon/pull/3645/files/ec02b5fe2c6835ffcd6cc1ecfa67c62f0b0d3d55#diff-2bc3f88bb67720bab520a61d6c56739b9d1a5239acbe0c290855e9a3d287ecdaR103 ?

Reply:
Nevermind then :)